wallet-extension/README.md

```diff
@@ -0,0 +1,112 @@
+# Arweave Web Wallet Extensions
+
+This file demonstrates how an Arweave Web Wallet Extension should structure it's API so that all dApps, as well as `arweave-js` can use it properly. An implementation of this can be found at [th8ta/ArConnect](https://github.com/th8ta/ArConnect).
+
+## Logic of the wallet
+
+Arweave web wallet extensions are not just simple middlewares to sign transactions. They allow the user to interact with dApps in a more seamless way, similar to how Web 2 logins work.
+
+A wallet extension should provide specific permissions for the API calls that the user has to approve. To do this, the extension should show a popup window on *connect*, to ask the user for these permissions. Ideally the wallet should also provide some sort of limitation of how much the specific dApp can spend from the user's wallet, because the popup won't show on each transaction signing.
+
+Preferably the wallet should notify the user in an understandable way every time a transaction signing occurs.
+
+## The global API object
+
+All web wallet extensions should provide (inject) a field inside the browsers `window` object called `arweaveWallet`. This can be accessed via `window.arweaveWallet`. An example `d.ts` file of this object can be found in [injected-api.d.ts](injected-api.d.ts).
+
+### Indicating the injection of the API object
+
+When injected, the extension should fire a custom event (`arweaveWalletLoaded`) for the current tab. A sample code for this can be found below, and should be added to the end of the injected script:
+
+```ts
+dispatchEvent(new CustomEvent("arweaveWalletLoaded", { detail: {} }));
+```
+
+Developers will be able to listen to this event this way:
+
+```ts
+window.addEventListener("arweaveWalletLoaded", () => {
+  /** Handle wallet extension loaded event **/
+});
+```
+
+### Wallet Switch Event
+
+If the extension provides multi-wallet support (the user is able to add more than 1 keyfiles to the extension), the wallet extension should always fire a custom event (`walletSwitch`), with the new address associated to the active keyfile as a parameter. The extension can choose to only fire the event if the app has the permissions `ACCESS_ADDRESS` or `ACCESS_ALL_ADDRESSES`.
+
+The event can be fired at the injected script:
+
+```ts
+dispatchEvent(new CustomEvent("walletSwitch", { detail: { address: newAddress } }));
+```
+
+Developers will be able to listen to this event this way:
+
+```ts
+window.addEventListener("arweaveWalletLoaded", () => {
+  const newAddress = e.detail.address;
+  /** Handle wallet extension wallet switch event **/
+});
+```
+
+### Important API functions
+
+These functions are used by `arweave-js` and they should be added to the global API object. See [njected-api.d.ts](injected-api.d.ts) for more.
+
+#### `connect(permissions, appInfo?): Promise<void>`
+
+Connect to the extension with a list of permissions and an optional application info (with a name and a logo).
```

Comment:
We should add types to the arguments of these functions

Reply:
That is in the example declaration file tho